tests/unit/common/json_encode_tests.cpp

```diff
@@ -50,3 +50,62 @@ TEST_CASE("Escape len") {
     REQUIRE(jsonify_str_buffer("\n") == strlen("\\n") + 1);
     REQUIRE(jsonify_str_buffer_len("\0", 1) == strlen("\\u0000") + 1);
 }
+
+TEST_CASE("Unescape json") {
+    char json[] = "1\\\"\\\\a\\\"34\\f\\b5\\r\\n6\\n\\\\78\\t0";
+    int i = unescape_json_i(json, strlen(json));
+
+    INFO("json: " + std::string(json));
+    const char *expected = "1\"\\a\"34\f\b5\r\n6\n\\78\t0";
+    REQUIRE(i == 10);
+    REQUIRE(strncmp(json, expected, strlen(expected)) == 0);
+}
+
+TEST_CASE("Unescape empty json") {
+    char json[] = "";
+    int i = unescape_json_i(json, strlen(json));
+
+    INFO("json: " + std::string(json));
+    REQUIRE(i == 0);
+    REQUIRE(strcmp(json, "") == 0);
+}
+
+TEST_CASE("Nothing to unescape json") {
+    char json[] = "1234567890abcdefgh";
+    int i = unescape_json_i(json, strlen(json));
+
+    INFO("json: " + std::string(json));
+    const char *expected = "1234567890abcdefgh";
+    REQUIRE(i == 0);
+    REQUIRE(strncmp(json, expected, strlen(expected)) == 0);
+}
+
+TEST_CASE("Unescape only part of string") {
+    char json[] = "\\\"abc\\\\12345";
+    int i = unescape_json_i(json, 4);
+
+    INFO("json: " + std::string(json));
+    const char *expected = "\"ab";
+    REQUIRE(i == 1);
+    REQUIRE(strncmp(json, expected, 3) == 0);
+}
+
+TEST_CASE("null unescape") {
```

Comment:
The null is not the only thing that can be encoded as a „number“. Anything (including „nice“ printable characters) can. How hard would be supporting these (and testing)?